### DISCUSSION TITLE
Memory exhausted with clear()

### DISCUSSION THREAD

**Discussion by duzenko** [doctrine/doctrine2#10585] [Support Questions]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.8.2

#### Summary

I wrote a new utility script that I run in CLI. It processes many records.

```
    foreach( $contractIds as $contractId ) {
        $contract = $entityManager->find( LoanContract::class, $contractId );
        $result = processContract( $contract );
        $processingResult[ $result ]++;
        $entityManager->clear();
    }
```

#### Current behavior

It works as expected locally and on staging, but on production it runs out of memory (standard 256MB). 
I have spent much time trying to debug it but the only thing that helps seems to be commenting out the clear() call.

```
    foreach( $contractIds as $contractId ) {
        $contract = $entityManager->find( LoanContract::class, $contractId );
        $result = processContract( $contract );
        $processingResult[ $result ]++;
//      $entityManager->clear();
    }
```
This version does not exhaust memory but takes much longer to complete.

I would appreciate any insight in what's going on here.

Attached is the log from two invocations: with `clear()` and without.
![image](https://user-images.githubusercontent.com/1481807/225399352-a2c77185-34e1-4ba8-aa2b-0e7c3737e44b.png)

**Comment by derrabus** [comment]
Moving to support. Without clear steps to reproduce your problem, we cannot file this as a bug report. In your particular case, I don't have any information on what `processContract()` does.

> Version 2.8.2

That version is more than two years old. Can you please try to reproduce this issue with a more recent version?

**Reply by duzenko** [reply]
Is it possible to disable entity cache so that I don't need to call `clear()` at all? Especially since it's only slowing things down.

**Reply by derrabus** [reply]
That's not a cache that you're clearing with that `clear()` call. But again, I need more information on what you're doing inside `processContract()` and on the structure of your entities.

**Reply by duzenko** [reply]
There's a bunch of business logic that reads and writes entities. Nothing 'funny'.

**Reply by derrabus** [reply]
Sorry, I'm not gonna ask you a third time. I cannot help you.

**Comment by duzenko** [comment]
FWIW my problem was different data on local and production - some records had huge blobs there.
